src/parseexpr.jl

```diff
@@ -272,7 +272,7 @@ addtoexpr(ex::AbstractArray{T}, c::AbstractArray, x::AbstractArray) where {T<:Ge
 addtoexpr(ex::AbstractArray{T}, c::AbstractArray, x::Number) where {T<:GenericAffExpr} = append!.(ex, c*x)
 addtoexpr(ex::AbstractArray{T}, c::Number, x::AbstractArray) where {T<:GenericAffExpr} = append!.(ex, c*x)
 
-addtoexpr(ex, c, x) = ex + c*x
+addtoexpr(ex, c, x) = ex .+ c*x
```

Comment:
Why should we be doing this by default instead of asking the user to use `.+` instead of `+` when appropriate?

Reply:
If I understand it correctly this is used in `parseExpr` by JuMP for constraints, e.g. `@SDconstraint Q >= eye(2)` is transformed into `addtoexpr(addtoexpr(zero(AffExpr), 1.0, Q), -1.0, eye(2))` so this is not due to a `+` in the user code

Reply:
Let's fix that then

Reply:
What should be done instead ?

Reply:
In some cases I already switched from `zero(AffExpr)` to `Val{false}()` as the default "zero" value: https://github.com/JuliaOpt/JuMP.jl/blob/110fbb64693f3f51f990cc43b0a34863c25609d0/src/macros.jl#L768

Reply:
Indeed, I have been able to remove my changes to parseexpr by doing that :)